src/Microsoft.Restier.Publishers.OData/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web.OData.Formatter.Deserialization;
+using Microsoft.OData.Core;
+using Microsoft.OData.Edm;
+using Microsoft.Restier.Core;
+
+namespace Microsoft.Restier.Publishers.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Get clr type from payload.
+    /// </summary>
+    internal static class DeserializationHelpers
+    {
+        internal static async Task<object> ConvertValueAsync(object oDataValue, Type expectedReturnType, IEdmTypeReference propertyType, ApiBase api)
```

Comment:
Do we have a public API to use from Web API OData instead of writing it by ourselves?